### DISCUSSION TITLE
Non-working bluetooth mouse side buttons

### DISCUSSION THREAD

**Discussion by dasef1eld** [blueman-project/blueman#2215] [Q&A] [answered]
blueman:
BlueZ:
Distribution:
Desktop environment:

* [YES ] I have consulted the [Troubleshooting page](https://github.com/blueman-project/blueman/wiki/Troubleshooting) and done my best effort to follow.

![image](https://github.com/blueman-project/blueman/assets/111562311/b9bb278c-92de-457f-8a01-ccd1321f211c)
Non-working side buttons (forward and backward) of the mi silent mouse bluetooth mouse in Linux mint. In Kubuntu and KDE Neon, these buttons work if connected via an adapter from the mouse, the buttons work, unfortunately not when connected via Bluetooth
MY PC [System info](https://pastebin.com/0gRFx7a2)

**Comment by cschramm** [comment, marked as answer]
The device is broken, see https://askubuntu.com/questions/1302323/back-forward-mouse-buttons-dont-show-up-in-evtest, referring to https://github.com/matega/hid_mimouse as a workaround.

**Reply by dasef1eld** [reply]
Thank you! [It's working!](https://askubuntu.com/questions/1302323/back-forward-mouse-buttons-dont-show-up-in-evtest)

> The mouse sends a bad HID report descriptor on Bluetooth (it sends the right one over USB). I wrote a kernel module to fix it: https://github.com/matega/hid_mimouse

Installation instructions:

sudo apt install git build-essential dkms
git clone https://github.com/matega/hid_mimouse.git
cd hid_mimouse
make
If make runs without errors you can

sudo make install
sudo dkms add .
sudo dkms install hid_mimouse/1.0
And reboot.
